src/validators/user-info-request-validator.ts

```diff
@@ -0,0 +1,68 @@
+import { UserInfoRequestError } from "../errors/user-info-request-error";
+import { IncomingHttpHeaders } from "http";
+import { signedJwtValidator } from "./signed-jwt-validator";
+import { Config } from "../config";
+import { logger } from "../logger";
+
+const AuthorisationHeaderKey: string = "authorization";
+const ValidScopes: string[] = ["openid", "email", "phone"];
+
+export const userInfoRequestValidator = async (
+  userInfoRequestHeaders: IncomingHttpHeaders
+): Promise<
+  | { valid: true }
+  | {
+      valid: false;
+      error: UserInfoRequestError;
+    }
+> => {
+  for (const key of Object.keys(userInfoRequestHeaders)) {
+    userInfoRequestHeaders[key.toLowerCase()] = userInfoRequestHeaders[key];
+  }
+  const accessToken = userInfoRequestHeaders[AuthorisationHeaderKey];
+  let match;
+  try {
+    match = /^Bearer (?<token>.*)$/.exec(accessToken as string);
+  } catch (error) {
+    logger.error("Failed To parse authorisation header.", error);
+    return { valid: false, error: UserInfoRequestError.MISSING_TOKEN };
+  }
+
+  if (!match?.groups?.token) {
+    logger.warn("Missing access token in authorisation header.");
+    return { valid: false, error: UserInfoRequestError.MISSING_TOKEN };
+  }
+
```

Comment:
I think we're missing expiration date check here

Reply:
The validator actually checks for expiration. There's a case for it in the user info tests. It would technically approve a token with no _exp_ field, but only if we had signed it with such a omission.

Reply:
It looks like if _exp_ were missing in a token given to our production endpoint we would actually exit the handler with an exception. I could add a check and return a 500, but I don't know if we really want to emulate that.

Reply:
Ah okay sorry I missed that - ideally we would emulate the entire system, but I think this feels like such an unusual case that I'm happy to leave it for now